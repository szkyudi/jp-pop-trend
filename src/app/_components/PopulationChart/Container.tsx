import { Presenter } from './Presenter.'
import { usePopulationChart } from './usePopuationChart'
import { Populations } from '@/lib/types/populations'

type Props = {
  populations: Populations
}
export function Container({ populations }: Props) {
  const { dataList } = usePopulationChart(populations)

  return <Presenter dataList={dataList} />
}
