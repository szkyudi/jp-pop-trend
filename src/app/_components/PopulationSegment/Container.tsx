'use client'

import { Presenter } from './Presenter'
import { usePopulationSegment } from './usePopulationSegment'

export function Container() {
  const { segmentItems } = usePopulationSegment()

  return <Presenter items={segmentItems} />
}
