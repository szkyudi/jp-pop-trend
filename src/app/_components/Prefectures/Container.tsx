'use client'

import { PrefecturesPresenter } from './Presenter'
import { components } from '@/lib/api/schema'
import { createContext, useState } from 'react'

export const CheckedPrefecturesContext = createContext<number[]>([])

type Props = {
  prefectures: components['schemas']['Prefecture'][]
  defaultCheckedPrefectures?: number[]
}
export function PrefecturesContainer({
  prefectures,
  defaultCheckedPrefectures = [],
}: Props) {
  const [checkedPrefectures, setCheckedPrefectures] = useState<number[]>(
    defaultCheckedPrefectures,
  )

  const handleChange = (prefCode: number) => (checked: boolean) => {
    if (checked) {
      setCheckedPrefectures([...checkedPrefectures, prefCode])
    } else {
      setCheckedPrefectures(
        checkedPrefectures.filter((code) => code !== prefCode),
      )
    }
  }

  return (
    <CheckedPrefecturesContext.Provider value={checkedPrefectures}>
      <PrefecturesPresenter
        prefectures={prefectures}
        onChange={handleChange}
        checkedPrefectures={checkedPrefectures}
      />
    </CheckedPrefecturesContext.Provider>
  )
}
