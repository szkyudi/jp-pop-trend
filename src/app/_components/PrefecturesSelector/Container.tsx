'use client'

import { Presenter } from './Presenter'
import {
  useCheckedPrefCodes,
  useSetCheckedPrefCodes,
} from '@/app/_contexts/checkedPrefCodes'

import { components } from '@/lib/api/schema'

type Props = {
  prefectures: components['schemas']['Prefecture'][]
}
export function Container({ prefectures }: Props) {
  const checkedPrefectures = useCheckedPrefCodes()
  const setCheckedPrefCodes = useSetCheckedPrefCodes()

  const handleChange = (prefCode: number) => (checked: boolean) => {
    if (checked) {
      setCheckedPrefCodes([...checkedPrefectures, prefCode])
    } else {
      setCheckedPrefCodes(
        checkedPrefectures.filter((code) => code !== prefCode),
      )
    }
  }

  return (
    <Presenter
      prefectures={prefectures}
      onChange={handleChange}
      checkedPrefectures={checkedPrefectures}
    />
  )
}
