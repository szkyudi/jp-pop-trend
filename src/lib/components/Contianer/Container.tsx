import s from './Container.module.scss'
import { ComponentPropsWithoutRef } from 'react'

type Props = ComponentPropsWithoutRef<'div'>
export function Container({ children }: Props) {
  return <div className={s.container}>{children}</div>
}
