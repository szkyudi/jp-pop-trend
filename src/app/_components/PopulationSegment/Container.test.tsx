import * as stories from './Container.stories'
import { render } from '@testing-library/react'
import { composeStories } from '@storybook/react'
import userEvent from '@testing-library/user-event'

const { Default } = composeStories(stories)

it('デフォルトで「総人口」が選択されていること', () => {
  const { getByRole } = render(<Default />)
  const radio = getByRole('radio', { name: '総人口' })

  expect(radio).toBeChecked()
})

it('選択肢が切り替わること', async () => {
  const user = userEvent.setup()

  const { getByRole } = render(<Default />)
  const defaultCheckedRadio = getByRole('radio', { name: '総人口' })
  const targetRadio = getByRole('radio', { name: '年少人口' })

  expect(defaultCheckedRadio).toBeChecked()

  await user.click(targetRadio)

  expect(targetRadio).toBeChecked()
  expect(defaultCheckedRadio).not.toBeChecked()
})
