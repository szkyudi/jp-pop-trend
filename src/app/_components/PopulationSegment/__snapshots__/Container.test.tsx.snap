// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`チェック時の表示が変わっていないこと 1`] = `
<div>
  <div
    class="_container_3e5575"
  >
    <fieldset
      role="radiogroup"
    >
      <ul
        class="_list_af03ec"
      >
        <li
          class="_item_37d96a"
        >
          <label
            class="_label_37d96a"
          >
            <input
              class="_input_37d96a"
              type="radio"
            />
            <span
              class="_text_37d96a"
            >
              総人口
            </span>
          </label>
        </li>
        <li
          class="_item_37d96a"
        >
          <label
            class="_label_37d96a"
          >
            <input
              class="_input_37d96a"
              type="radio"
            />
            <span
              class="_text_37d96a"
            >
              年少人口
            </span>
          </label>
        </li>
        <li
          class="_item_37d96a"
        >
          <label
            class="_label_37d96a"
          >
            <input
              class="_input_37d96a"
              type="radio"
            />
            <span
              class="_text_37d96a"
            >
              生産年齢人口
            </span>
          </label>
        </li>
        <li
          class="_item_37d96a"
        >
          <label
            class="_label_37d96a"
          >
            <input
              class="_input_37d96a"
              type="radio"
            />
            <span
              class="_text_37d96a"
            >
              老年人口
            </span>
          </label>
        </li>
      </ul>
    </fieldset>
  </div>
</div>
`;

exports[`デフォルトの表示が変わっていないこと 1`] = `
<div>
  <div
    class="_container_3e5575"
  >
    <fieldset
      role="radiogroup"
    >
      <ul
        class="_list_af03ec"
      >
        <li
          class="_item_37d96a"
        >
          <label
            class="_label_37d96a"
          >
            <input
              class="_input_37d96a"
              type="radio"
            />
            <span
              class="_text_37d96a"
            >
              総人口
            </span>
          </label>
        </li>
        <li
          class="_item_37d96a"
        >
          <label
            class="_label_37d96a"
          >
            <input
              class="_input_37d96a"
              type="radio"
            />
            <span
              class="_text_37d96a"
            >
              年少人口
            </span>
          </label>
        </li>
        <li
          class="_item_37d96a"
        >
          <label
            class="_label_37d96a"
          >
            <input
              class="_input_37d96a"
              type="radio"
            />
            <span
              class="_text_37d96a"
            >
              生産年齢人口
            </span>
          </label>
        </li>
        <li
          class="_item_37d96a"
        >
          <label
            class="_label_37d96a"
          >
            <input
              class="_input_37d96a"
              type="radio"
            />
            <span
              class="_text_37d96a"
            >
              老年人口
            </span>
          </label>
        </li>
      </ul>
    </fieldset>
  </div>
</div>
`;

exports[`フォーカス時の表示が変わっていないこと 1`] = `
<div>
  <div
    class="_container_3e5575"
  >
    <fieldset
      role="radiogroup"
    >
      <ul
        class="_list_af03ec"
      >
        <li
          class="_item_37d96a"
        >
          <label
            class="_label_37d96a"
          >
            <input
              class="_input_37d96a"
              type="radio"
            />
            <span
              class="_text_37d96a"
            >
              総人口
            </span>
          </label>
        </li>
        <li
          class="_item_37d96a"
        >
          <label
            class="_label_37d96a"
          >
            <input
              class="_input_37d96a"
              type="radio"
            />
            <span
              class="_text_37d96a"
            >
              年少人口
            </span>
          </label>
        </li>
        <li
          class="_item_37d96a"
        >
          <label
            class="_label_37d96a"
          >
            <input
              class="_input_37d96a"
              type="radio"
            />
            <span
              class="_text_37d96a"
            >
              生産年齢人口
            </span>
          </label>
        </li>
        <li
          class="_item_37d96a"
        >
          <label
            class="_label_37d96a"
          >
            <input
              class="_input_37d96a"
              type="radio"
            />
            <span
              class="_text_37d96a"
            >
              老年人口
            </span>
          </label>
        </li>
      </ul>
    </fieldset>
  </div>
</div>
`;
